# Zone Analysis

This section of the analysis looks that the probabilities surrounding different zones and application selections.


In [1]:
import pandas as pd

# Import cleaned data from csv file
df = pd.read_csv(
    "./2021_results_cleaned.csv",
    # Import was failing to parse date columns, so I
    # to add the column names
    parse_dates=[
        "preferred_entry_date_1",
        "preferred_entry_date_2",
        "preferred_entry_date_3",
        "awarded_entry_date",
    ],
    date_format="%m-%d-%Y",  # Align format with export format
    na_filter=False,  # Do not convert 'N/A' to NaN
)

df.head()

preferred_entry_date_1   preferred_division_1  \
0             2021-07-25  Core Enchantment Zone   
1             2021-08-12  Core Enchantment Zone   
2             2021-07-30  Core Enchantment Zone   
3             2021-07-01  Core Enchantment Zone   
4             2021-06-21          Colchuck Zone   

   minimum_acceptable_group_size_1 preferred_entry_date_2  \
0                                7             2021-08-01   
1                                8             2021-08-12   
2                                4             2021-08-14   
3                                4             2021-06-09   
4                                2             2021-06-28   

    preferred_division_2  minimum_acceptable_group_size_2  \
0  Core Enchantment Zone                                7   
1          Colchuck Zone                                8   
2  Core Enchantment Zone                                4   
3              Snow Zone                                4   
4          Colchuck Zone                                2   

  preferred_entry_date_3     preferred_division_3  \
0             2021-07-25                Snow Zone   
1             2021-08-12  Eightmile/Caroline Zone   
2             2021-07-16    Core Enchantment Zone   
3             2021-07-07            Colchuck Zone   
4             2021-07-13             Stuart  Zone   

   minimum_acceptable_group_size_3 results_status  awarded_preference  \
0                                8   Unsuccessful                   0   
1                                8   Unsuccessful                   0   
2                                7   Unsuccessful                   0   
3                                4   Unsuccessful                   0   
4                                2   Unsuccessful                   0   

  awarded_entry_date awarded_entrance_code_name  awarded_group_size  
0         1970-01-01                        N/A                   0  
1         1970-01-01                        N/A                   0  
2         1970-01-01                        N/A                   0  
3         1970-01-01                        N/A                   0  
4         1970-01-01                        N/A                   0

In [2]:
total = len(df)
print(f"Total number of applications in 2021: {total}")

Total number of applications in 2021: 36695


In [3]:
# Simple probability of being awarded a permit
accepted = df["results_status"] == "Accepted"
no_response = df["results_status"] == "No Response"
declined = df["results_status"] == "Declined"

awarded_filter = accepted | no_response | declined

awarded = df[accepted | no_response | declined]

total_awarded = len(awarded)

print(f"Total number of awarded applications in 2021: {total_awarded}")

Total number of awarded applications in 2021: 2445


In [4]:
prob_awarded = total_awarded / total

print(
    f"Probability of being awarded a permit in 2021: {prob_awarded:.2%} ({total_awarded}/{total})"
)

Probability of being awarded a permit in 2021: 6.66% (2445/36695)


In [5]:
# Which zone was awarded the most permits?
pd.crosstab(
    awarded["awarded_entrance_code_name"],
    awarded["results_status"],
    margins=True,
)

results_status                   Accepted  Declined  No Response   All
awarded_entrance_code_name                                            
Colchuck Zone                         298         6           44   348
Core Enchantment Zone                 595         8           70   673
Eightmile/Caroline Zone               224        10           60   294
Eightmile/Caroline Zone (stock)        15         0            5    20
Snow Zone                             536        20           86   642
Stuart  Zone                          355        13           86   454
Stuart Zone (stock)                    13         0            1    14
All                                  2036        57          352  2445

In [6]:
# Event A - You are awarded a permit for the Core Enchantments

awarded_core = awarded["awarded_entrance_code_name"] == "Core Enchantment Zone"
total_awarded_core = len(awarded[awarded_core])

prob_awarded_core = total_awarded_core / total

# You are awarded a permit for the Core Enchantments
print(
    f"Probability that you were awarded a permit for the Core Enchantments: {prob_awarded_core:.2%} ({total_awarded_core}/{total})"
)

Probability that you were awarded a permit for the Core Enchantments: 1.83% (673/36695)


In [7]:
# Event A - You applied for a permit for the Core Enchantments
# Event B - You were awarded a permit

# Given that you applied for a permit for the Core Enchantments,
# What is the probability that you were awarded a permit?

applied_core_1 = df["preferred_division_1"] == "Core Enchantment Zone"
applied_core_2 = df["preferred_division_2"] == "Core Enchantment Zone"
applied_core_3 = df["preferred_division_3"] == "Core Enchantment Zone"

applied_core_filter = applied_core_1 | applied_core_2 | applied_core_3

applied_core = df[applied_core_filter]

total_applied_core = len(applied_core)

prob_applied_core = total_applied_core / total

total_applied_core_awarded = len(df[applied_core_filter & awarded_filter])

prob_applied_core_awarded = total_applied_core_awarded / total_applied_core

# P(B|A) = P(B and A) / P(A)
prob_awarded_given_applied_core = prob_applied_core_awarded / prob_applied_core

print(
    f"Probability you were awarded a permit given you applied for the Core Enchantments Zone: {prob_awarded_given_applied_core:.2%} ({prob_applied_core_awarded}/{prob_applied_core})"
)

Probability you were awarded a permit given you applied for the Core Enchantments Zone: 6.76% (0.056166136281618054/0.8306581278103283)


In [8]:
# Event B - You were awarded a permit
# Event A - You did not apply for a permit for the Core Enchantments

# Given that you did not apply for a permit for the Core Enchantments,
# what is the probability that you were awarded a permit?

did_not_apply_core_filter = ~applied_core_filter

did_not_apply_core = df[did_not_apply_core_filter]

total_did_not_apply_core = len(did_not_apply_core)  # 6214

prob_did_not_apply_core = total_did_not_apply_core / total  # 6214 / 36695 = 0.1694

total_did_not_apply_core_awarded = len(
    df[did_not_apply_core_filter & awarded_filter]
)  # 733

prob_did_not_apply_core_awarded = (
    total_did_not_apply_core_awarded / total
)  # 733 / 36695 = 0.0199

# P(B|A) = P(B and A) / P(A)
prob_awarded_given_did_not_apply_core = (
    prob_did_not_apply_core_awarded / prob_did_not_apply_core
)

print(
    f"Probability you were awarded a permit given you did not apply for the Core Enchantments Zone: {prob_awarded_given_did_not_apply_core:.2%} ({prob_did_not_apply_core_awarded}/{prob_did_not_apply_core})"
)

Probability you were awarded a permit given you did not apply for the Core Enchantments Zone: 11.80% (0.019975473497751736/0.16934187218967162)


|                        | Awarded | Not Awarded |       |
| ---------------------- | ------- | ----------- | ----- |
| Applied for Core       | 1712    | 28769       | 30481 |
| Did not apply for Core | 733     | 5481        | 6214  |
|                        | 2445    | 34250       | 36695 |

There was an 11.80% you were awarded a permit given you **did not** apply for a Core Enchantments Permit.

There was an 6.76% you were awarded a permit given you **did** apply for a Core Enchantments Permit.


In [9]:
awarded_core_filter = df["awarded_entrance_code_name"] == "Core Enchantment Zone"
division_2_na_filter = df["preferred_division_2"] == "N/A"
division_3_na_filter = df["preferred_division_3"] == "N/A"

only_applied_core_filter = applied_core_1 & applied_core_2 & applied_core_3
only_applied_core_2_filter = (
    applied_core_1 & division_2_na_filter & division_3_na_filter
)
only_applied_core_3_filter = applied_core_1 & applied_core_2 & division_3_na_filter

only_applied_core = df[
    only_applied_core_filter | only_applied_core_2_filter | only_applied_core_3_filter
]

total_only_applied_core = len(only_applied_core)

total_only_applied_core_awarded_core = len(
    df[only_applied_core_filter & awarded_core_filter]
)

prob_only_applied_core_awarded_core = (
    total_only_applied_core_awarded_core / total_only_applied_core
)

print(
    f"Probability you were awarded a permit for the Core Enchantments given you only applied for a Core Enchantments Permit: {prob_only_applied_core_awarded_core:.2%} ({total_only_applied_core_awarded_core}/{total_only_applied_core})"
)

Probability you were awarded a permit for the Core Enchantments given you only applied for a Core Enchantments Permit: 2.63% (467/17743)


In [10]:
def sort_zone_probabilities(x):
    return x.sort(key=lambda x: x[1], reverse=True)


def create_zone_probability_dataframe(x, columns):
    return pd.DataFrame(
        x,
        columns=columns,
    )


def add_probability_percent_column(df):
    df["Probability (%)"] = df["Probability"].map("{:.2%}".format)
    return df


def zone_probabilities_to_crosstab(x, columns):
    sort_zone_probabilities(x)
    return add_probability_percent_column(create_zone_probability_dataframe(x, columns))

In [11]:
# Get a list of all the zones
zones_values = df["preferred_division_1"].unique()

prob_awarded_zone_applied_for = []

# Loop over the zones and calculate the probability of being awarded a permit for each zone
for zone in zones_values:
    applied_1 = df["preferred_division_1"] == zone
    applied_2 = df["preferred_division_2"] == zone
    applied_3 = df["preferred_division_3"] == zone

    zone_filter = applied_1 | applied_2 | applied_3

    applied_zone = df[zone_filter]

    total_zone = len(applied_zone)

    awarded_zone_filter = df["awarded_entrance_code_name"] == zone

    total_zone_awarded = len(df[zone_filter & awarded_zone_filter])
    prob_zone_awarded = total_zone_awarded / total_zone

    prob_awarded_zone_applied_for.append(
        [zone, prob_zone_awarded, total_zone_awarded, total_zone]
    )

    print(
        f"Probability of being awarded a permit for {zone}, given applied 1+ option in zone: {prob_zone_awarded:.2%} ({total_zone_awarded}/{total_zone})"
    )


df_prob_awarded_zone_applied_for = zone_probabilities_to_crosstab(
    prob_awarded_zone_applied_for,
    ["Zone", "Probability", "Total Awarded", "Total Applied"],
)

# Show crosstab of the new dataframe
df_prob_awarded_zone_applied_for

Probability of being awarded a permit for Core Enchantment Zone, given applied 1+ option in zone: 2.21% (673/30481)
Probability of being awarded a permit for Colchuck Zone, given applied 1+ option in zone: 2.97% (348/11698)
Probability of being awarded a permit for Snow Zone, given applied 1+ option in zone: 7.46% (642/8607)
Probability of being awarded a permit for Stuart  Zone, given applied 1+ option in zone: 10.09% (454/4501)
Probability of being awarded a permit for Eightmile/Caroline Zone, given applied 1+ option in zone: 14.56% (294/2019)
Probability of being awarded a permit for Eightmile/Caroline Zone (stock), given applied 1+ option in zone: 8.81% (20/227)
Probability of being awarded a permit for Stuart Zone (stock), given applied 1+ option in zone: 7.95% (14/176)


Zone  Probability  Total Awarded  Total Applied  \
0          Eightmile/Caroline Zone     0.145617            294           2019   
1                     Stuart  Zone     0.100866            454           4501   
2  Eightmile/Caroline Zone (stock)     0.088106             20            227   
3              Stuart Zone (stock)     0.079545             14            176   
4                        Snow Zone     0.074590            642           8607   
5                    Colchuck Zone     0.029749            348          11698   
6            Core Enchantment Zone     0.022079            673          30481   

  Probability (%)  
0          14.56%  
1          10.09%  
2           8.81%  
3           7.95%  
4           7.46%  
5           2.97%  
6           2.21%

# Split Entries

This part of the analysis splits the application into separate entries. It then adds columns for day of the week, month, and whether or not the application entry was awarded or not.


In [12]:
df.columns

Index(['preferred_entry_date_1', 'preferred_division_1',
       'minimum_acceptable_group_size_1', 'preferred_entry_date_2',
       'preferred_division_2', 'minimum_acceptable_group_size_2',
       'preferred_entry_date_3', 'preferred_division_3',
       'minimum_acceptable_group_size_3', 'results_status',
       'awarded_preference', 'awarded_entry_date',
       'awarded_entrance_code_name', 'awarded_group_size'],
      dtype='object')

In [13]:
# It may be better to break up each individual entry into its own row, so that the data can be analyzed more easily.
preferred_options = [1, 2, 3]

# Columns that every dataframe will have
shared_columns = [
    "results_status",
    "awarded_preference",
    "awarded_entry_date",
    "awarded_entrance_code_name",
    "awarded_group_size",
]
new_dataframes = []

# Iterate over each option number creating a new dataframe for each
for option in preferred_options:
    # Get the columns for the current option
    columns = [
        f"preferred_division_{option}",
        f"preferred_entry_date_{option}",
        f"minimum_acceptable_group_size_{option}",
    ]
    # Create a new dataframe for the current option
    df_option = df[columns + shared_columns].copy()
    # Rename the columns to remove the option number
    df_option.columns = [
        "preferred_division",
        "preferred_entry_date",
        "minimum_acceptable_group_size",
    ] + shared_columns
    # Add a column to indicate if the permit was awarded for the current option
    df_option["awarded"] = df_option["awarded_preference"] == option
    df_option["preferred_option"] = option

    # Append the new dataframe to the list of dataframes
    new_dataframes.append(df_option)

# Concatenate the list of dataframes into a single dataframe
df_split = pd.concat(new_dataframes)

# Check the new dataframe
df_split.head()

preferred_division preferred_entry_date  minimum_acceptable_group_size  \
0  Core Enchantment Zone           2021-07-25                              7   
1  Core Enchantment Zone           2021-08-12                              8   
2  Core Enchantment Zone           2021-07-30                              4   
3  Core Enchantment Zone           2021-07-01                              4   
4          Colchuck Zone           2021-06-21                              2   

  results_status  awarded_preference awarded_entry_date  \
0   Unsuccessful                   0         1970-01-01   
1   Unsuccessful                   0         1970-01-01   
2   Unsuccessful                   0         1970-01-01   
3   Unsuccessful                   0         1970-01-01   
4   Unsuccessful                   0         1970-01-01   

  awarded_entrance_code_name  awarded_group_size  awarded  preferred_option  
0                        N/A                   0    False                 1  
1                        N/A                   0    False                 1  
2                        N/A                   0    False                 1  
3                        N/A                   0    False                 1  
4                        N/A                   0    False                 1

In [14]:
# Drop rows where the preferred division is N/A
df_split = df_split[df_split["preferred_division"] != "N/A"]

# Check the new dataframe
df_split["preferred_division"].value_counts()

preferred_division
Core Enchantment Zone              70433
Colchuck Zone                      16493
Snow Zone                          12481
Stuart  Zone                        6346
Eightmile/Caroline Zone             2425
Eightmile/Caroline Zone (stock)      265
Stuart Zone (stock)                  199
Name: count, dtype: int64

In [15]:
# Add the month of the preferred entry date to the dataframe
import calendar

# Get the month as an integer
df_split["preferred_entry_date" + "_month"] = df_split["preferred_entry_date"].dt.month
# Get the month as a string
df_split["preferred_entry_date" + "_month"] = df_split[
    "preferred_entry_date" + "_month"
].apply(lambda x: calendar.month_name[x])

# Check the data
df_split.head()

preferred_division preferred_entry_date  minimum_acceptable_group_size  \
0  Core Enchantment Zone           2021-07-25                              7   
1  Core Enchantment Zone           2021-08-12                              8   
2  Core Enchantment Zone           2021-07-30                              4   
3  Core Enchantment Zone           2021-07-01                              4   
4          Colchuck Zone           2021-06-21                              2   

  results_status  awarded_preference awarded_entry_date  \
0   Unsuccessful                   0         1970-01-01   
1   Unsuccessful                   0         1970-01-01   
2   Unsuccessful                   0         1970-01-01   
3   Unsuccessful                   0         1970-01-01   
4   Unsuccessful                   0         1970-01-01   

  awarded_entrance_code_name  awarded_group_size  awarded  preferred_option  \
0                        N/A                   0    False                 1   
1                        N/A                   0    False                 1   
2                        N/A                   0    False                 1   
3                        N/A                   0    False                 1   
4                        N/A                   0    False                 1   

  preferred_entry_date_month  
0                       July  
1                     August  
2                       July  
3                       July  
4                       June

In [16]:
# Look at the month column and add up the amounts
df_split["preferred_entry_date_month"].value_counts()

preferred_entry_date_month
August       40877
July         30188
September    23842
June          8579
October       3518
May           1638
Name: count, dtype: int64

In [17]:
# Add the day of the week columns based on preferred entry date
df_split["preferred_entry_date" + "_day"] = df_split[
    "preferred_entry_date"
].dt.day_name()

# Check the data
df_split.head()

preferred_division preferred_entry_date  minimum_acceptable_group_size  \
0  Core Enchantment Zone           2021-07-25                              7   
1  Core Enchantment Zone           2021-08-12                              8   
2  Core Enchantment Zone           2021-07-30                              4   
3  Core Enchantment Zone           2021-07-01                              4   
4          Colchuck Zone           2021-06-21                              2   

  results_status  awarded_preference awarded_entry_date  \
0   Unsuccessful                   0         1970-01-01   
1   Unsuccessful                   0         1970-01-01   
2   Unsuccessful                   0         1970-01-01   
3   Unsuccessful                   0         1970-01-01   
4   Unsuccessful                   0         1970-01-01   

  awarded_entrance_code_name  awarded_group_size  awarded  preferred_option  \
0                        N/A                   0    False                 1   
1                        N/A                   0    False                 1   
2                        N/A                   0    False                 1   
3                        N/A                   0    False                 1   
4                        N/A                   0    False                 1   

  preferred_entry_date_month preferred_entry_date_day  
0                       July                   Sunday  
1                     August                 Thursday  
2                       July                   Friday  
3                       July                 Thursday  
4                       June                   Monday

In [18]:
# Look at the day of the week column and add up the amounts
df_split["preferred_entry_date_day"].value_counts()

preferred_entry_date_day
Friday       21918
Thursday     20072
Monday       15372
Wednesday    15112
Saturday     13766
Tuesday      13734
Sunday        8668
Name: count, dtype: int64

In [19]:
# Create a datatframe of the values we KNOW were skipped during
# the lottery process because the preference could not be accomodated.

awarded_preference_greater_than_0 = df_split["awarded_preference"] > 0

# Find where the preferred option was less than the awarded preference
preferred_option_less_than_awarded_option = (
    df_split["preferred_option"] < df_split["awarded_preference"]
)

df_split_skipped = df_split[
    awarded_preference_greater_than_0 & preferred_option_less_than_awarded_option
].copy()

df_split_skipped.head(5)

preferred_division preferred_entry_date  \
42   Core Enchantment Zone           2021-08-11   
65   Core Enchantment Zone           2021-09-06   
105              Snow Zone           2021-08-25   
134  Core Enchantment Zone           2021-07-04   
136  Core Enchantment Zone           2021-07-03   

     minimum_acceptable_group_size results_status  awarded_preference  \
42                               2       Accepted                   2   
65                               4       Accepted                   3   
105                              4       Accepted                   2   
134                              4       Accepted                   3   
136                              6    No Response                   3   

    awarded_entry_date awarded_entrance_code_name  awarded_group_size  \
42          2021-07-06    Eightmile/Caroline Zone                   2   
65          2021-09-19                  Snow Zone                   4   
105         2021-08-18                  Snow Zone                   4   
134         2021-07-11               Stuart  Zone                   4   
136         2021-10-09      Core Enchantment Zone                   6   

     awarded  preferred_option preferred_entry_date_month  \
42     False                 1                     August   
65     False                 1                  September   
105    False                 1                     August   
134    False                 1                       July   
136    False                 1                       July   

    preferred_entry_date_day  
42                 Wednesday  
65                    Monday  
105                Wednesday  
134                   Sunday  
136                 Saturday

## Zone


In [20]:
# Create an awarded to skipped comparison for the zones

prob_zone_skipped = []

for zone in zones_values:
    zone_filter = df_split_skipped["preferred_division"] == zone

    zone_data = df_split_skipped[zone_filter]

    total_zone = len(zone_data)

    total_skipped = len(df_split_skipped)

    prob_zone_df_split_skipped = total_zone / total_skipped

    prob_zone_skipped.append(
        [
            zone,
            prob_zone_df_split_skipped,
            total_skipped,
            total_zone,
        ]
    )

    print(
        f"Probability of being awarded but skipped given {zone}: {prob_zone_df_split_skipped:.2%} ({total_zone}/{total_skipped})"
    )

# Create dataframe from list
df_prob_zone_skipped = zone_probabilities_to_crosstab(
    prob_zone_skipped,
    ["Zone", "Probability", "Total Skipped", "Total Zone Skipped"],
)

# Show crosstab of the new dataframe
df_prob_zone_skipped

Probability of being awarded but skipped given Core Enchantment Zone: 61.58% (1146/1861)
Probability of being awarded but skipped given Colchuck Zone: 20.10% (374/1861)
Probability of being awarded but skipped given Snow Zone: 8.76% (163/1861)
Probability of being awarded but skipped given Stuart  Zone: 6.72% (125/1861)
Probability of being awarded but skipped given Eightmile/Caroline Zone: 2.20% (41/1861)
Probability of being awarded but skipped given Eightmile/Caroline Zone (stock): 0.38% (7/1861)
Probability of being awarded but skipped given Stuart Zone (stock): 0.27% (5/1861)


Zone  Probability  Total Skipped  \
0            Core Enchantment Zone     0.615798           1861   
1                    Colchuck Zone     0.200967           1861   
2                        Snow Zone     0.087587           1861   
3                     Stuart  Zone     0.067168           1861   
4          Eightmile/Caroline Zone     0.022031           1861   
5  Eightmile/Caroline Zone (stock)     0.003761           1861   
6              Stuart Zone (stock)     0.002687           1861   

   Total Zone Skipped Probability (%)  
0                1146          61.58%  
1                 374          20.10%  
2                 163           8.76%  
3                 125           6.72%  
4                  41           2.20%  
5                   7           0.38%  
6                   5           0.27%

In [21]:
prob_awarded_zone_split = []

# Loop over the zones and calculate the probability of being awarded a permit for each zone
for zone in zones_values:
    zone_filter = df_split["preferred_division"] == zone

    applied_zone = df_split[zone_filter]

    total_zone = len(applied_zone)

    awarded_zone_filter = df_split["awarded"] == True

    total_zone_awarded = len(df_split[zone_filter & awarded_zone_filter])
    prob_zone_awarded = total_zone_awarded / total_zone

    prob_awarded_zone_split.append(
        [zone, prob_zone_awarded, total_zone_awarded, total_zone]
    )

    print(
        f"Probability of being awarded a permit for {zone}: {prob_zone_awarded:.2%} ({total_zone_awarded}/{total_zone})"
    )

# Create dataframe from list
df_prob_awarded_zone_split = zone_probabilities_to_crosstab(
    prob_awarded_zone_split, ["Zone", "Probability", "Total Awarded", "Total Applied"]
)

# Probability of being awarded a permit
awarded_split_filter = df_split["awarded"] == True

awarded_split = df_split[awarded_split_filter]

total_awarded_split = len(awarded_split)

total_split = len(df_split)

prob_awarded_split = total_awarded_split / total_split

print(
    f"Probability of being awarded a permit by entry option: {prob_awarded_split:.2%} ({total_awarded_split}/{total_split})"
)

# Show crosstab of the new dataframe
df_prob_awarded_zone_split

Probability of being awarded a permit for Core Enchantment Zone: 0.96% (673/70433)
Probability of being awarded a permit for Colchuck Zone: 2.11% (348/16493)
Probability of being awarded a permit for Snow Zone: 5.14% (642/12481)
Probability of being awarded a permit for Stuart  Zone: 7.15% (454/6346)
Probability of being awarded a permit for Eightmile/Caroline Zone: 12.12% (294/2425)
Probability of being awarded a permit for Eightmile/Caroline Zone (stock): 7.55% (20/265)
Probability of being awarded a permit for Stuart Zone (stock): 7.04% (14/199)
Probability of being awarded a permit by entry option: 2.25% (2445/108642)


Zone  Probability  Total Awarded  Total Applied  \
0          Eightmile/Caroline Zone     0.121237            294           2425   
1  Eightmile/Caroline Zone (stock)     0.075472             20            265   
2                     Stuart  Zone     0.071541            454           6346   
3              Stuart Zone (stock)     0.070352             14            199   
4                        Snow Zone     0.051438            642          12481   
5                    Colchuck Zone     0.021100            348          16493   
6            Core Enchantment Zone     0.009555            673          70433   

  Probability (%)  
0          12.12%  
1           7.55%  
2           7.15%  
3           7.04%  
4           5.14%  
5           2.11%  
6           0.96%

In [22]:
# Create an awarded to skipped ratio for zones
# Get the total and zone columns from the zone skipped dataframe
df_total_skipped_by_zone = df_prob_zone_skipped[["Total Zone Skipped", "Zone"]]
df_total_awarded_by_zone = df_prob_awarded_zone_split[["Total Awarded", "Zone"]]

# Merge the total and zone columns with the awarded group size dataframe
df_prob_zone_skipped_total = pd.merge(
    df_total_skipped_by_zone, df_total_awarded_by_zone, on="Zone"
)

# Calculate the percentage of awarded group size entries compared to the skipped group size entries
df_prob_zone_skipped_total["Ratio"] = (
    df_prob_zone_skipped_total["Total Awarded"]
    / df_prob_zone_skipped_total["Total Zone Skipped"]
)

# Sort the dataframe by the Ratio column
df_prob_zone_skipped_total = df_prob_zone_skipped_total.sort_values(
    by="Ratio", ascending=False
)

# Show the Ratio column as a Ratio
df_prob_zone_skipped_total["Ratio"] = df_prob_zone_skipped_total["Ratio"].map(
    "{:.2f}".format
)

# Reorder columns
df_prob_zone_skipped_total = df_prob_zone_skipped_total[
    [
        "Zone",
        "Total Awarded",
        "Total Zone Skipped",
        "Ratio",
    ]
]

# Show the dataframe
df_prob_zone_skipped_total

Zone  Total Awarded  Total Zone Skipped Ratio
4          Eightmile/Caroline Zone            294                  41  7.17
2                        Snow Zone            642                 163  3.94
3                     Stuart  Zone            454                 125  3.63
5  Eightmile/Caroline Zone (stock)             20                   7  2.86
6              Stuart Zone (stock)             14                   5  2.80
1                    Colchuck Zone            348                 374  0.93
0            Core Enchantment Zone            673                1146  0.59

## Day


In [23]:
# The awarded but split data frame is a collection of entries that we know were passed over in the lottery
# because they were awarded a permit for a different option than they preferred.

# What is the probability your option was passed over given the day of the week?

days_of_week = df_split_skipped["preferred_entry_date_day"].unique()

prob_skipped_day = []

for day in days_of_week:
    day_filter = df_split_skipped["preferred_entry_date_day"] == day

    day_data = df_split_skipped[day_filter]

    total_day_skipped = len(day_data)

    total_skipped = len(df_split_skipped)

    prob_day_skipped = total_day_skipped / total_skipped

    prob_skipped_day.append([day, prob_day_skipped, total_skipped, total_day_skipped])

    print(
        f"Probability of being awarded but skipped given the day of the week is {day}: {prob_day_skipped:.2%} ({total_day_skipped}/{total_skipped})"
    )

# Create dataframe from list
df_prob_skipped_day = zone_probabilities_to_crosstab(
    prob_skipped_day,
    ["Day", "Probability", "Total Skipped", "Total Day Skipped"],
)

# Show crosstab of the new dataframe
df_prob_skipped_day

Probability of being awarded but skipped given the day of the week is Wednesday: 12.47% (232/1861)
Probability of being awarded but skipped given the day of the week is Monday: 15.21% (283/1861)
Probability of being awarded but skipped given the day of the week is Sunday: 12.09% (225/1861)
Probability of being awarded but skipped given the day of the week is Saturday: 14.02% (261/1861)
Probability of being awarded but skipped given the day of the week is Friday: 17.84% (332/1861)
Probability of being awarded but skipped given the day of the week is Thursday: 14.67% (273/1861)
Probability of being awarded but skipped given the day of the week is Tuesday: 13.70% (255/1861)


Day  Probability  Total Skipped  Total Day Skipped Probability (%)
0     Friday     0.178399           1861                332          17.84%
1     Monday     0.152069           1861                283          15.21%
2   Thursday     0.146695           1861                273          14.67%
3   Saturday     0.140247           1861                261          14.02%
4    Tuesday     0.137023           1861                255          13.70%
5  Wednesday     0.124664           1861                232          12.47%
6     Sunday     0.120903           1861                225          12.09%

In [24]:
# Awarded day of the week
split_awarded = df_split[(df_split["awarded"] == True)]

prob_split_day = []

# Loop over the group sizes on Sunday in August and see who did the best
for day in days_of_week:
    split_day_filter = split_awarded["preferred_entry_date_day"] == day

    total_split_awarded = len(split_awarded)

    split_day_awarded = split_awarded[split_day_filter]

    total_split_day_awarded = len(split_day_awarded)

    prob_split_awarded = total_split_day_awarded / total_split_awarded

    print(
        f"Probability of being awarded a permit on {day}: {prob_split_awarded:.2%} ({total_split_day_awarded}/{total_split_awarded})"
    )

    prob_split_day.append(
        [
            day,
            prob_split_awarded,
            total_split_day_awarded,
            total_split_awarded,
        ]
    )

# Create dataframe from list
df_prob_split_day = zone_probabilities_to_crosstab(
    prob_split_day,
    [
        "Day",
        "Probability",
        "Total Awarded Day of Week",
        "Total Awarded",
    ],
)

# Show crosstab of the new dataframe
df_prob_split_day

Probability of being awarded a permit on Wednesday: 13.50% (330/2445)
Probability of being awarded a permit on Monday: 13.50% (330/2445)
Probability of being awarded a permit on Sunday: 17.71% (433/2445)
Probability of being awarded a permit on Saturday: 14.36% (351/2445)
Probability of being awarded a permit on Friday: 13.42% (328/2445)
Probability of being awarded a permit on Thursday: 13.70% (335/2445)
Probability of being awarded a permit on Tuesday: 13.82% (338/2445)


Day  Probability  Total Awarded Day of Week  Total Awarded  \
0     Sunday     0.177096                        433           2445   
1   Saturday     0.143558                        351           2445   
2    Tuesday     0.138241                        338           2445   
3   Thursday     0.137014                        335           2445   
4  Wednesday     0.134969                        330           2445   
5     Monday     0.134969                        330           2445   
6     Friday     0.134151                        328           2445   

  Probability (%)  
0          17.71%  
1          14.36%  
2          13.82%  
3          13.70%  
4          13.50%  
5          13.50%  
6          13.42%

In [25]:
# Calculate the awarded to skipped for day of week overall
df_total_day_awarded = df_prob_split_day[["Day", "Total Awarded Day of Week"]]
df_total_day_skipped = df_prob_skipped_day[["Day", "Total Day Skipped"]]

df_day_awarded_skipped = pd.merge(df_total_day_awarded, df_total_day_skipped, on="Day")

df_day_awarded_skipped["Awarded to Skipped"] = (
    df_day_awarded_skipped["Total Awarded Day of Week"]
    / df_day_awarded_skipped["Total Day Skipped"]
)

# Sort Values
df_day_awarded_skipped = df_day_awarded_skipped.sort_values(
    by="Awarded to Skipped", ascending=False
)

# Show as probability
df_day_awarded_skipped["Awarded to Skipped"] = df_day_awarded_skipped[
    "Awarded to Skipped"
].map("{:.2f}".format)


df_day_awarded_skipped

Day  Total Awarded Day of Week  Total Day Skipped Awarded to Skipped
0     Sunday                        433                225               1.92
4  Wednesday                        330                232               1.42
1   Saturday                        351                261               1.34
2    Tuesday                        338                255               1.33
3   Thursday                        335                273               1.23
5     Monday                        330                283               1.17
6     Friday                        328                332               0.99

## Month


In [26]:
# What is the probability your option was passed over given the month?
months = df_split_skipped["preferred_entry_date_month"].unique()

prob_skipped_month = []

for month in months:
    month_filter = df_split_skipped["preferred_entry_date_month"] == month

    month_data = df_split_skipped[month_filter]

    total_month_skipped = len(month_data)

    total_skipped = len(df_split_skipped)

    prob_month_df_split_skipped = total_month_skipped / total_skipped

    prob_skipped_month.append(
        [month, prob_month_df_split_skipped, total_skipped, total_month_skipped]
    )

    print(
        f"Probability of being awarded but skipped given the month is {month}: {prob_month_df_split_skipped:.2%} ({total_month_skipped}/{total_skipped})"
    )

# Create dataframe from list
df_prob_skipped_month = zone_probabilities_to_crosstab(
    prob_skipped_month,
    ["Month", "Probability", "Total Awarded But Skipped", "Total Month Skipped"],
)

# Show crosstab of the new dataframe
df_prob_skipped_month

Probability of being awarded but skipped given the month is August: 26.11% (486/1861)
Probability of being awarded but skipped given the month is September: 24.99% (465/1861)
Probability of being awarded but skipped given the month is July: 23.43% (436/1861)
Probability of being awarded but skipped given the month is May: 4.46% (83/1861)
Probability of being awarded but skipped given the month is October: 8.81% (164/1861)
Probability of being awarded but skipped given the month is June: 12.20% (227/1861)


Month  Probability  Total Awarded But Skipped  Total Month Skipped  \
0     August     0.261150                       1861                  486   
1  September     0.249866                       1861                  465   
2       July     0.234283                       1861                  436   
3       June     0.121977                       1861                  227   
4    October     0.088125                       1861                  164   
5        May     0.044600                       1861                   83   

  Probability (%)  
0          26.11%  
1          24.99%  
2          23.43%  
3          12.20%  
4           8.81%  
5           4.46%

In [27]:
# Get total awarded by month
split_awarded = df_split[(df_split["awarded"] == True)]

prob_split_month = []

# Loop over the group sizes on Sunday in August and see who did the best
for month in months:
    split_month_filter = split_awarded["preferred_entry_date_month"] == month

    total_split_awarded = len(split_awarded)

    split_month_awarded = split_awarded[split_month_filter]

    total_split_month_awarded = len(split_month_awarded)

    prob_split_awarded = total_split_month_awarded / total_split_awarded

    print(
        f"Probability of being awarded a permit in {month}: {prob_split_awarded:.2%} ({total_split_month_awarded}/{total_split_awarded})"
    )

    prob_split_month.append(
        [
            month,
            prob_split_awarded,
            total_split_month_awarded,
            total_split_awarded,
        ]
    )

# Create dataframe from list
df_prob_split_month = zone_probabilities_to_crosstab(
    prob_split_month,
    [
        "Month",
        "Probability",
        "Total Awarded By Month",
        "Total Awarded",
    ],
)

# Show crosstab of the new dataframe
df_prob_split_month

Probability of being awarded a permit in August: 19.88% (486/2445)
Probability of being awarded a permit in September: 18.73% (458/2445)
Probability of being awarded a permit in July: 19.59% (479/2445)
Probability of being awarded a permit in May: 9.94% (243/2445)
Probability of being awarded a permit in October: 12.92% (316/2445)
Probability of being awarded a permit in June: 18.94% (463/2445)


Month  Probability  Total Awarded By Month  Total Awarded  \
0     August     0.198773                     486           2445   
1       July     0.195910                     479           2445   
2       June     0.189366                     463           2445   
3  September     0.187321                     458           2445   
4    October     0.129243                     316           2445   
5        May     0.099387                     243           2445   

  Probability (%)  
0          19.88%  
1          19.59%  
2          18.94%  
3          18.73%  
4          12.92%  
5           9.94%

In [28]:
# Get total awarded by month
df_total_awarded_month = df_prob_split_month[["Total Awarded By Month", "Month"]]
df_total_skipped_month = df_prob_skipped_month[["Total Month Skipped", "Month"]]

# Merge the data frames
df_total_awarded_month_skipped = pd.merge(
    df_total_awarded_month, df_total_skipped_month, on="Month"
)

# Calculate the awarded to skipped ratio
df_total_awarded_month_skipped["Ratio"] = (
    df_total_awarded_month_skipped["Total Awarded By Month"]
    / df_total_awarded_month_skipped["Total Month Skipped"]
)

# Sort the values
df_total_awarded_month_skipped = df_total_awarded_month_skipped.sort_values(
    by="Ratio", ascending=False
)

# Show as ratio
df_total_awarded_month_skipped["Ratio"] = df_total_awarded_month_skipped["Ratio"].map(
    "{:.2f}".format
)

# Sort the columns
df_total_awarded_month_skipped = df_total_awarded_month_skipped[
    [
        "Month",
        "Total Awarded By Month",
        "Total Month Skipped",
        "Ratio",
    ]
]

# Show the data frame
df_total_awarded_month_skipped

Month  Total Awarded By Month  Total Month Skipped Ratio
5        May                     243                   83  2.93
2       June                     463                  227  2.04
4    October                     316                  164  1.93
1       July                     479                  436  1.10
0     August                     486                  486  1.00
3  September                     458                  465  0.98

## Core Zone


In [29]:
# Next look at these same statistics but for the Core Enchantment Zone
split_core_zone_filter = (
    df_split_skipped["preferred_division"] == "Core Enchantment Zone"
)

df_split_skipped_core_zone = df_split_skipped[split_core_zone_filter]

df_split_skipped_core_zone

preferred_division preferred_entry_date  \
42     Core Enchantment Zone           2021-08-11   
65     Core Enchantment Zone           2021-09-06   
134    Core Enchantment Zone           2021-07-04   
136    Core Enchantment Zone           2021-07-03   
209    Core Enchantment Zone           2021-07-01   
...                      ...                  ...   
36411  Core Enchantment Zone           2021-10-10   
36469  Core Enchantment Zone           2021-06-23   
36532  Core Enchantment Zone           2021-09-26   
36640  Core Enchantment Zone           2021-10-04   
36661  Core Enchantment Zone           2021-09-26   

       minimum_acceptable_group_size results_status  awarded_preference  \
42                                 2       Accepted                   2   
65                                 4       Accepted                   3   
134                                4       Accepted                   3   
136                                6    No Response                   3   
209                                8       Accepted                   2   
...                              ...            ...                 ...   
36411                              4       Accepted                   3   
36469                              2       Accepted                   3   
36532                              4       Accepted                   3   
36640                              6       Accepted                   3   
36661                              6       Accepted                   3   

      awarded_entry_date awarded_entrance_code_name  awarded_group_size  \
42            2021-07-06    Eightmile/Caroline Zone                   2   
65            2021-09-19                  Snow Zone                   4   
134           2021-07-11               Stuart  Zone                   4   
136           2021-10-09      Core Enchantment Zone                   6   
209           2021-07-01              Colchuck Zone                   8   
...                  ...                        ...                 ...   
36411         2021-10-17      Core Enchantment Zone                   4   
36469         2021-06-13      Core Enchantment Zone                   2   
36532         2021-09-26                  Snow Zone                   6   
36640         2021-10-28      Core Enchantment Zone                   6   
36661         2021-10-03      Core Enchantment Zone                   6   

       awarded  preferred_option preferred_entry_date_month  \
42       False                 1                     August   
65       False                 1                  September   
134      False                 1                       July   
136      False                 1                       July   
209      False                 1                       July   
...        ...               ...                        ...   
36411    False                 2                    October   
36469    False                 2                       June   
36532    False                 2                  September   
36640    False                 2                    October   
36661    False                 2                  September   

      preferred_entry_date_day  
42                   Wednesday  
65                      Monday  
134                     Sunday  
136                   Saturday  
209                   Thursday  
...                        ...  
36411                   Sunday  
36469                Wednesday  
36532                   Sunday  
36640                   Monday  
36661                   Sunday  

[1146 rows x 12 columns]

### Core Zone Month


In [30]:
prob_skipped_core_zone_month = []

for month in months:
    month_filter = df_split_skipped_core_zone["preferred_entry_date_month"] == month

    month_data = df_split_skipped_core_zone[month_filter]

    total_month_skipped_core_zone = len(month_data)

    total_skipped_core_zone = len(df_split_skipped_core_zone)

    prob_month_awarded_but_skipped_core_zone = (
        total_month_skipped_core_zone / total_skipped_core_zone
    )

    prob_skipped_core_zone_month.append(
        [
            month,
            prob_month_awarded_but_skipped_core_zone,
            total_skipped_core_zone,
            total_month_skipped_core_zone,
        ]
    )

    print(
        f"Probability of being awarded but skipped given the month is {month}: {prob_month_awarded_but_skipped_core_zone:.2%} ({total_month_skipped_core_zone}/{total_skipped_core_zone})"
    )

# Create dataframe from list
df_prob_skipped_core_zone_month = zone_probabilities_to_crosstab(
    prob_skipped_core_zone_month,
    [
        "Month",
        "Probability",
        "Total Skipped Core Zone",
        "Total Month Skipped Core Zone",
    ],
)

# Show crosstab of the new dataframe
df_prob_skipped_core_zone_month

Probability of being awarded but skipped given the month is August: 25.39% (291/1146)
Probability of being awarded but skipped given the month is September: 24.87% (285/1146)
Probability of being awarded but skipped given the month is July: 22.43% (257/1146)
Probability of being awarded but skipped given the month is May: 5.06% (58/1146)
Probability of being awarded but skipped given the month is October: 9.95% (114/1146)
Probability of being awarded but skipped given the month is June: 12.30% (141/1146)


Month  Probability  Total Skipped Core Zone  \
0     August     0.253927                     1146   
1  September     0.248691                     1146   
2       July     0.224258                     1146   
3       June     0.123037                     1146   
4    October     0.099476                     1146   
5        May     0.050611                     1146   

   Total Month Skipped Core Zone Probability (%)  
0                            291          25.39%  
1                            285          24.87%  
2                            257          22.43%  
3                            141          12.30%  
4                            114           9.95%  
5                             58           5.06%

In [31]:
# Get all the Core Enchantment Zone application entries
split_core_zone_filter = df_split["preferred_division"] == "Core Enchantment Zone"

df_split_core_zone = df_split[split_core_zone_filter]

df_split_core_zone

preferred_division preferred_entry_date  \
0      Core Enchantment Zone           2021-07-25   
1      Core Enchantment Zone           2021-08-12   
2      Core Enchantment Zone           2021-07-30   
3      Core Enchantment Zone           2021-07-01   
5      Core Enchantment Zone           2021-08-06   
...                      ...                  ...   
36684  Core Enchantment Zone           2021-08-10   
36687  Core Enchantment Zone           2021-07-20   
36689  Core Enchantment Zone           2021-09-03   
36692  Core Enchantment Zone           2021-08-14   
36694  Core Enchantment Zone           2021-08-05   

       minimum_acceptable_group_size results_status  awarded_preference  \
0                                  7   Unsuccessful                   0   
1                                  8   Unsuccessful                   0   
2                                  4   Unsuccessful                   0   
3                                  4   Unsuccessful                   0   
5                                  4   Unsuccessful                   0   
...                              ...            ...                 ...   
36684                              6   Unsuccessful                   0   
36687                              4   Unsuccessful                   0   
36689                              6   Unsuccessful                   0   
36692                              2   Unsuccessful                   0   
36694                              6   Unsuccessful                   0   

      awarded_entry_date awarded_entrance_code_name  awarded_group_size  \
0             1970-01-01                        N/A                   0   
1             1970-01-01                        N/A                   0   
2             1970-01-01                        N/A                   0   
3             1970-01-01                        N/A                   0   
5             1970-01-01                        N/A                   0   
...                  ...                        ...                 ...   
36684         1970-01-01                        N/A                   0   
36687         1970-01-01                        N/A                   0   
36689         1970-01-01                        N/A                   0   
36692         1970-01-01                        N/A                   0   
36694         1970-01-01                        N/A                   0   

       awarded  preferred_option preferred_entry_date_month  \
0        False                 1                       July   
1        False                 1                     August   
2        False                 1                       July   
3        False                 1                       July   
5        False                 1                     August   
...        ...               ...                        ...   
36684    False                 3                     August   
36687    False                 3                       July   
36689    False                 3                  September   
36692    False                 3                     August   
36694    False                 3                     August   

      preferred_entry_date_day  
0                       Sunday  
1                     Thursday  
2                       Friday  
3                     Thursday  
5                       Friday  
...                        ...  
36684                  Tuesday  
36687                  Tuesday  
36689                   Friday  
36692                 Saturday  
36694                 Thursday  

[70433 rows x 12 columns]

In [32]:
# Probability awarded by month for the Core Enchantment Zone
split_core_zone_awarded = df_split_core_zone[(df_split_core_zone["awarded"] == True)]

prob_split_core_zone_month = []

months = df_split_core_zone["preferred_entry_date_month"].unique()

# Loop over the group sizes on Sunday in August and see who did the best
for month in months:
    split_core_zone_month_filter = (
        split_core_zone_awarded["preferred_entry_date_month"] == month
    )

    total_split_core_zone_awarded = len(split_core_zone_awarded)

    split_core_zone_month_awarded = split_core_zone_awarded[
        split_core_zone_month_filter
    ]

    total_split_core_zone_month_awarded = len(split_core_zone_month_awarded)

    prob_split_core_zone_awarded = (
        total_split_core_zone_month_awarded / total_split_core_zone_awarded
    )

    print(
        f"Probability of being awarded a permit for the Core Enchantments on {month}: {prob_split_core_zone_awarded:.2%} ({total_split_core_zone_month_awarded}/{total_split_core_zone_awarded})"
    )

    prob_split_core_zone_month.append(
        [
            month,
            prob_split_core_zone_awarded,
            total_split_core_zone_month_awarded,
            total_split_core_zone_awarded,
        ]
    )

# Create dataframe from list
df_prob_split_core_zone_month = zone_probabilities_to_crosstab(
    prob_split_core_zone_month,
    [
        "Month",
        "Probability",
        "Total Awarded Month Core Zone",
        "Total Awarded Core Zone",
    ],
)

# Show crosstab of the new dataframe
df_prob_split_core_zone_month

Probability of being awarded a permit for the Core Enchantments on July: 18.13% (122/673)
Probability of being awarded a permit for the Core Enchantments on August: 18.57% (125/673)
Probability of being awarded a permit for the Core Enchantments on June: 18.13% (122/673)
Probability of being awarded a permit for the Core Enchantments on September: 16.94% (114/673)
Probability of being awarded a permit for the Core Enchantments on October: 16.94% (114/673)
Probability of being awarded a permit for the Core Enchantments on May: 11.29% (76/673)


Month  Probability  Total Awarded Month Core Zone  \
0     August     0.185736                            125   
1       July     0.181278                            122   
2       June     0.181278                            122   
3  September     0.169391                            114   
4    October     0.169391                            114   
5        May     0.112927                             76   

   Total Awarded Core Zone Probability (%)  
0                      673          18.57%  
1                      673          18.13%  
2                      673          18.13%  
3                      673          16.94%  
4                      673          16.94%  
5                      673          11.29%

In [33]:
# Let's do the same analysis look at the month
# Get the total and months columns from the month skipped dataframe
df_total_skipped_by_month = df_prob_skipped_core_zone_month[
    ["Total Month Skipped Core Zone", "Month"]
]
df_total_awarded_by_month = df_prob_split_core_zone_month[
    ["Total Awarded Month Core Zone", "Month"]
]

# Merge the total and month columns with the awarded group size dataframe
df_prob_month_skipped_core_zone_total = pd.merge(
    df_total_skipped_by_month, df_total_awarded_by_month, on="Month"
)

# Calculate the percentage of awarded group size entries compared to the skipped group size entries
df_prob_month_skipped_core_zone_total["Ratio"] = (
    df_prob_month_skipped_core_zone_total["Total Awarded Month Core Zone"]
    / df_prob_month_skipped_core_zone_total["Total Month Skipped Core Zone"]
)

# Sort the dataframe by the percentage column
df_prob_month_skipped_core_zone_total = (
    df_prob_month_skipped_core_zone_total.sort_values(by="Ratio", ascending=False)
)

# Show the ratio column to two decimal places
df_prob_month_skipped_core_zone_total["Ratio"] = df_prob_month_skipped_core_zone_total[
    "Ratio"
].map("{:.2f}".format)

# Reorder columns
df_prob_month_skipped_core_zone_total = df_prob_month_skipped_core_zone_total[
    [
        "Month",
        "Total Awarded Month Core Zone",
        "Total Month Skipped Core Zone",
        "Ratio",
    ]
]

# Show the dataframe
df_prob_month_skipped_core_zone_total

Month  Total Awarded Month Core Zone  Total Month Skipped Core Zone  \
5        May                             76                             58   
4    October                            114                            114   
3       June                            122                            141   
2       July                            122                            257   
0     August                            125                            291   
1  September                            114                            285   

  Ratio  
5  1.31  
4  1.00  
3  0.87  
2  0.47  
0  0.43  
1  0.40

### Core Zone Day


In [34]:
prob_skipped_core_zone_day = []

for day in days_of_week:
    day_filter = df_split_skipped_core_zone["preferred_entry_date_day"] == day

    day_data = df_split_skipped_core_zone[day_filter]

    total_day_skipped_core_zone = len(day_data)

    total_skipped_core_zone = len(df_split_skipped_core_zone)

    prob_day_df_split_skipped_core_zone = (
        total_day_skipped_core_zone / total_skipped_core_zone
    )

    prob_skipped_core_zone_day.append(
        [
            day,
            prob_day_df_split_skipped_core_zone,
            total_skipped_core_zone,
            total_day_skipped_core_zone,
        ]
    )

    print(
        f"Probability of being awarded but skipped given the day is {day}: {prob_day_df_split_skipped_core_zone:.2%} ({total_day_skipped_core_zone}/{total_skipped_core_zone})"
    )

# Create dataframe from list
df_prob_skipped_core_zone_day = zone_probabilities_to_crosstab(
    prob_skipped_core_zone_day,
    ["Day", "Probability", "Total Skipped Core Zone", "Total Skipped by Day Core Zone"],
)

# Show crosstab of the new dataframe
df_prob_skipped_core_zone_day

Probability of being awarded but skipped given the day is Wednesday: 13.44% (154/1146)
Probability of being awarded but skipped given the day is Monday: 16.06% (184/1146)
Probability of being awarded but skipped given the day is Sunday: 12.57% (144/1146)
Probability of being awarded but skipped given the day is Saturday: 12.91% (148/1146)
Probability of being awarded but skipped given the day is Friday: 16.49% (189/1146)
Probability of being awarded but skipped given the day is Thursday: 15.01% (172/1146)
Probability of being awarded but skipped given the day is Tuesday: 13.53% (155/1146)


Day  Probability  Total Skipped Core Zone  \
0     Friday     0.164921                     1146   
1     Monday     0.160558                     1146   
2   Thursday     0.150087                     1146   
3    Tuesday     0.135253                     1146   
4  Wednesday     0.134380                     1146   
5   Saturday     0.129145                     1146   
6     Sunday     0.125654                     1146   

   Total Skipped by Day Core Zone Probability (%)  
0                             189          16.49%  
1                             184          16.06%  
2                             172          15.01%  
3                             155          13.53%  
4                             154          13.44%  
5                             148          12.91%  
6                             144          12.57%

In [35]:
split_core_zone_awarded = df_split_core_zone[(df_split_core_zone["awarded"] == True)]

prob_split_core_zone_day = []

# Loop over the group sizes on Sunday in August and see who did the best
for day in days_of_week:
    split_core_zone_day_filter = (
        split_core_zone_awarded["preferred_entry_date_day"] == day
    )

    total_split_core_zone_awarded = len(split_core_zone_awarded)

    split_core_zone_day_awarded = split_core_zone_awarded[split_core_zone_day_filter]

    total_split_core_zone_day_awarded = len(split_core_zone_day_awarded)

    prob_split_core_zone_awarded = (
        total_split_core_zone_day_awarded / total_split_core_zone_awarded
    )

    print(
        f"Probability of being awarded a permit for the Core Enchantments on {day}: {prob_split_core_zone_awarded:.2%} ({total_split_core_zone_day_awarded}/{total_split_core_zone_awarded})"
    )

    prob_split_core_zone_day.append(
        [
            day,
            prob_split_core_zone_awarded,
            total_split_core_zone_day_awarded,
            total_split_core_zone_awarded,
        ]
    )

# Create dataframe from list
df_prob_split_core_zone_day = zone_probabilities_to_crosstab(
    prob_split_core_zone_day,
    [
        "Day",
        "Probability",
        "Total Awarded Day of Week Core Zone",
        "Total Awarded Core Zone",
    ],
)

# Show crosstab of the new dataframe
df_prob_split_core_zone_day

Probability of being awarded a permit for the Core Enchantments on Wednesday: 13.22% (89/673)
Probability of being awarded a permit for the Core Enchantments on Monday: 12.63% (85/673)
Probability of being awarded a permit for the Core Enchantments on Sunday: 18.87% (127/673)
Probability of being awarded a permit for the Core Enchantments on Saturday: 14.41% (97/673)
Probability of being awarded a permit for the Core Enchantments on Friday: 12.48% (84/673)
Probability of being awarded a permit for the Core Enchantments on Thursday: 13.82% (93/673)
Probability of being awarded a permit for the Core Enchantments on Tuesday: 14.56% (98/673)


Day  Probability  Total Awarded Day of Week Core Zone  \
0     Sunday     0.188707                                  127   
1    Tuesday     0.145617                                   98   
2   Saturday     0.144131                                   97   
3   Thursday     0.138187                                   93   
4  Wednesday     0.132244                                   89   
5     Monday     0.126300                                   85   
6     Friday     0.124814                                   84   

   Total Awarded Core Zone Probability (%)  
0                      673          18.87%  
1                      673          14.56%  
2                      673          14.41%  
3                      673          13.82%  
4                      673          13.22%  
5                      673          12.63%  
6                      673          12.48%

In [36]:
# Let's do the same analysis look but for day of the week and month
# Get the total and day columns from the day skipped dataframe
df_total_skipped_by_day = df_prob_skipped_core_zone_day[
    ["Total Skipped by Day Core Zone", "Day"]
]
df_total_awarded_by_day = df_prob_split_core_zone_day[
    ["Total Awarded Day of Week Core Zone", "Day"]
]

# Merge the total and day columns with the awarded group size dataframe
df_prob_day_skipped_core_zone_total = pd.merge(
    df_total_skipped_by_day, df_total_awarded_by_day, on="Day"
)

# Calculate the percentage of awarded group size entries compared to the skipped group size entries
df_prob_day_skipped_core_zone_total["Ratio"] = (
    df_prob_day_skipped_core_zone_total["Total Awarded Day of Week Core Zone"]
    / df_prob_day_skipped_core_zone_total["Total Skipped by Day Core Zone"]
)

# Sort the dataframe by the Ratio column
df_prob_day_skipped_core_zone_total = df_prob_day_skipped_core_zone_total.sort_values(
    by="Ratio", ascending=False
)

# Show the Ratio column as a Ratio
df_prob_day_skipped_core_zone_total["Ratio"] = df_prob_day_skipped_core_zone_total[
    "Ratio"
].map("{:.2f}".format)

# Reorder columns
df_prob_day_skipped_core_zone_total = df_prob_day_skipped_core_zone_total[
    [
        "Day",
        "Total Awarded Day of Week Core Zone",
        "Total Skipped by Day Core Zone",
        "Ratio",
    ]
]

# Show the dataframe
df_prob_day_skipped_core_zone_total

Day  Total Awarded Day of Week Core Zone  \
6     Sunday                                  127   
5   Saturday                                   97   
3    Tuesday                                   98   
4  Wednesday                                   89   
2   Thursday                                   93   
1     Monday                                   85   
0     Friday                                   84   

   Total Skipped by Day Core Zone Ratio  
6                             144  0.88  
5                             148  0.66  
3                             155  0.63  
4                             154  0.58  
2                             172  0.54  
1                             184  0.46  
0                             189  0.44

### Core Zone Group Size


In [37]:
# Look at the group size value counts for the core zone
df_split_core_zone["minimum_acceptable_group_size"].value_counts()

minimum_acceptable_group_size
4    19677
8    16341
6    12720
2    11306
5     4973
3     3887
7      885
1      644
Name: count, dtype: int64

In [38]:
# Create filter for awarded core zone
awarded_core_zone_filter = df_split_core_zone["awarded"] == True

# Get the awarded core zone entries
awarded_core_zone = df_split_core_zone[awarded_core_zone_filter]

# Look at the group size value counts for the awarded core zone
awarded_core_zone["minimum_acceptable_group_size"].value_counts()

minimum_acceptable_group_size
4    191
2    164
6    101
8     98
1     42
3     36
5     35
7      6
Name: count, dtype: int64

In [39]:
group_sizes = df_split_skipped_core_zone["minimum_acceptable_group_size"].unique()

prob_group_size_skipped_core_zone = []

for group_size in group_sizes:
    group_size_filter = (
        df_split_skipped_core_zone["minimum_acceptable_group_size"] == group_size
    )

    group_size_data = df_split_skipped_core_zone[group_size_filter]

    total_group_size = len(group_size_data)

    total_skipped_core_zone = len(df_split_skipped_core_zone)

    prob_group_size_df_split_skipped_core_zone = (
        total_group_size / total_skipped_core_zone
    )

    prob_group_size_skipped_core_zone.append(
        [
            group_size,
            prob_group_size_df_split_skipped_core_zone,
            total_skipped_core_zone,
            total_group_size,
        ]
    )

    print(
        f"Probability of being awarded but skipped given the minimum acceptable group size is {group_size} in Core Zone: {prob_group_size_df_split_skipped_core_zone:.2%} ({total_group_size}/{total_skipped_core_zone})"
    )

# Create dataframe from list
df_prob_group_size_skipped_core_zone = zone_probabilities_to_crosstab(
    prob_group_size_skipped_core_zone,
    [
        "Group Size",
        "Probability",
        "Total Skipped Core Zone",
        "Total Skipped Group Size Core Zone",
    ],
)

# Show crosstab of the new dataframe
df_prob_group_size_skipped_core_zone

Probability of being awarded but skipped given the minimum acceptable group size is 2 in Core Zone: 20.42% (234/1146)
Probability of being awarded but skipped given the minimum acceptable group size is 4 in Core Zone: 30.28% (347/1146)
Probability of being awarded but skipped given the minimum acceptable group size is 6 in Core Zone: 14.57% (167/1146)
Probability of being awarded but skipped given the minimum acceptable group size is 8 in Core Zone: 18.76% (215/1146)
Probability of being awarded but skipped given the minimum acceptable group size is 1 in Core Zone: 2.79% (32/1146)
Probability of being awarded but skipped given the minimum acceptable group size is 5 in Core Zone: 7.16% (82/1146)
Probability of being awarded but skipped given the minimum acceptable group size is 3 in Core Zone: 4.36% (50/1146)
Probability of being awarded but skipped given the minimum acceptable group size is 7 in Core Zone: 1.66% (19/1146)


Group Size  Probability  Total Skipped Core Zone  \
0           4     0.302792                     1146   
1           2     0.204188                     1146   
2           8     0.187609                     1146   
3           6     0.145724                     1146   
4           5     0.071553                     1146   
5           3     0.043630                     1146   
6           1     0.027923                     1146   
7           7     0.016579                     1146   

   Total Skipped Group Size Core Zone Probability (%)  
0                                 347          30.28%  
1                                 234          20.42%  
2                                 215          18.76%  
3                                 167          14.57%  
4                                  82           7.16%  
5                                  50           4.36%  
6                                  32           2.79%  
7                                  19           1.66%

In [40]:
# Awarded permits for the core zone by group size
split_core_zone_awarded = df_split_core_zone[(df_split_core_zone["awarded"] == True)]

prob_split_core_zone_group_size = []

# Loop over the group sizes on Sunday in August and see who did the best
for group in group_sizes:
    split_core_zone_group_size_filter = (
        split_core_zone_awarded["minimum_acceptable_group_size"] == group
    )

    total_split_core_zone_awarded = len(split_core_zone_awarded)

    split_core_zone_group_awarded = split_core_zone_awarded[
        split_core_zone_group_size_filter
    ]

    total_split_core_zone_group_awarded = len(split_core_zone_group_awarded)

    prob_split_core_zone_awarded = (
        total_split_core_zone_group_awarded / total_split_core_zone_awarded
    )

    print(
        f"Probability of being awarded a permit for the Core Enchantments with a group size of {group}: {prob_split_core_zone_awarded:.2%} ({total_split_core_zone_group_awarded}/{total_split_core_zone_awarded})"
    )

    prob_split_core_zone_group_size.append(
        [
            group,
            prob_split_core_zone_awarded,
            total_split_core_zone_group_awarded,
            total_split_core_zone_awarded,
        ]
    )

# Create dataframe from list
df_prob_split_core_zone_group_size = zone_probabilities_to_crosstab(
    prob_split_core_zone_group_size,
    [
        "Group Size",
        "Probability",
        "Total Awarded Group Size Core Zone",
        "Total Awarded Core Zone",
    ],
)

# Show crosstab of the new dataframe
df_prob_split_core_zone_group_size

Probability of being awarded a permit for the Core Enchantments with a group size of 2: 24.37% (164/673)
Probability of being awarded a permit for the Core Enchantments with a group size of 4: 28.38% (191/673)
Probability of being awarded a permit for the Core Enchantments with a group size of 6: 15.01% (101/673)
Probability of being awarded a permit for the Core Enchantments with a group size of 8: 14.56% (98/673)
Probability of being awarded a permit for the Core Enchantments with a group size of 1: 6.24% (42/673)
Probability of being awarded a permit for the Core Enchantments with a group size of 5: 5.20% (35/673)
Probability of being awarded a permit for the Core Enchantments with a group size of 3: 5.35% (36/673)
Probability of being awarded a permit for the Core Enchantments with a group size of 7: 0.89% (6/673)


Group Size  Probability  Total Awarded Group Size Core Zone  \
0           4     0.283804                                 191   
1           2     0.243685                                 164   
2           6     0.150074                                 101   
3           8     0.145617                                  98   
4           1     0.062407                                  42   
5           3     0.053492                                  36   
6           5     0.052006                                  35   
7           7     0.008915                                   6   

   Total Awarded Core Zone Probability (%)  
0                      673          28.38%  
1                      673          24.37%  
2                      673          15.01%  
3                      673          14.56%  
4                      673           6.24%  
5                      673           5.35%  
6                      673           5.20%  
7                      673           0.89%

In [41]:
# Compare the skipped group size entries to the awarded group size entries in the Core Enchantment Zone
# Get the total and group size columns from the group size skipped dataframe
df_total_skipped_by_group_size = df_prob_group_size_skipped_core_zone[
    ["Total Skipped Group Size Core Zone", "Group Size"]
]
df_total_awarded_by_group_size = df_prob_split_core_zone_group_size[
    ["Total Awarded Group Size Core Zone", "Group Size"]
]

# Merge the total and group size columns with the awarded group size dataframe
df_prob_group_size_skipped_core_zone_total = pd.merge(
    df_total_skipped_by_group_size, df_total_awarded_by_group_size, on="Group Size"
)

# Calculate the percentage of awarded group size entries compared to the skipped group size entries
df_prob_group_size_skipped_core_zone_total["Ratio"] = (
    df_prob_group_size_skipped_core_zone_total["Total Awarded Group Size Core Zone"]
    / df_prob_group_size_skipped_core_zone_total["Total Skipped Group Size Core Zone"]
)

# Sort the dataframe by the Ratio column
df_prob_group_size_skipped_core_zone_total = (
    df_prob_group_size_skipped_core_zone_total.sort_values(by="Ratio", ascending=False)
)

# Show the Ratio column as a ratio
df_prob_group_size_skipped_core_zone_total["Ratio"] = (
    df_prob_group_size_skipped_core_zone_total["Ratio"].map("{:.2f}".format)
)

# Reorder columns
df_prob_group_size_skipped_core_zone_total = df_prob_group_size_skipped_core_zone_total[
    [
        "Group Size",
        "Total Awarded Group Size Core Zone",
        "Total Skipped Group Size Core Zone",
        "Ratio",
    ]
]

# Show the dataframe
df_prob_group_size_skipped_core_zone_total

Group Size  Total Awarded Group Size Core Zone  \
6           1                                  42   
5           3                                  36   
1           2                                 164   
3           6                                 101   
0           4                                 191   
2           8                                  98   
4           5                                  35   
7           7                                   6   

   Total Skipped Group Size Core Zone Ratio  
6                                  32  1.31  
5                                  50  0.72  
1                                 234  0.70  
3                                 167  0.60  
0                                 347  0.55  
2                                 215  0.46  
4                                  82  0.43  
7                                  19  0.32

In [42]:
split_core_zone_august_filter = (
    df_split_core_zone["preferred_entry_date_month"] == "August"
)

split_core_zone_august = df_split_core_zone[split_core_zone_august_filter]

split_core_zone_august_awarded_filter = split_core_zone_august["awarded"] == True

split_core_zone_august_sunday_filter = (
    split_core_zone_august["preferred_entry_date_day"] == "Sunday"
)
split_core_zone_august_monday_filter = (
    split_core_zone_august["preferred_entry_date_day"] == "Monday"
)
split_core_zone_august_tuesday_filter = (
    split_core_zone_august["preferred_entry_date_day"] == "Tuesday"
)

split_core_zone_august_friday_filter = (
    split_core_zone_august["preferred_entry_date_day"] == "Friday"
)
split_core_zone_august_saturday_filter = (
    split_core_zone_august["preferred_entry_date_day"] == "Saturday"
)
split_core_zone_august_thursday_filter = (
    split_core_zone_august["preferred_entry_date_day"] == "Thursday"
)

prob_split_core_zone_august_sunday_group_size = []

split_core_zone_august_sunday_awarded = split_core_zone_august[
    split_core_zone_august_sunday_filter & (split_core_zone_august["awarded"] == True)
]

In [43]:
# Print the total number of people that entered the Core Zone on Saturdays in August
august_saturday_total = split_core_zone_august[
    split_core_zone_august_saturday_filter & split_core_zone_august_awarded_filter
]["minimum_acceptable_group_size"].sum()
print(
    f"Total number of people awarded entrance in the Core Zone on Saturdays in August: {august_saturday_total}"
)

# Print the total number of people that entered the Core Zone on Sundays in August
august_sunday_total = split_core_zone_august[
    split_core_zone_august_sunday_filter & split_core_zone_august_awarded_filter
]["minimum_acceptable_group_size"].sum()
print(
    f"Total number of people awarded entrance in the Core Zone on Sundays in August: {august_sunday_total}"
)

# Print the total number of people that entered the Core Zone on Mondays in August
august_monday_total = split_core_zone_august[
    split_core_zone_august_monday_filter & split_core_zone_august_awarded_filter
]["minimum_acceptable_group_size"].sum()

print(
    f"Total number of people awarded entrance in the Core Zone on Mondays in August: {august_monday_total}"
)

Total number of people awarded entrance in the Core Zone on Saturdays in August: 63
Total number of people awarded entrance in the Core Zone on Sundays in August: 119
Total number of people awarded entrance in the Core Zone on Mondays in August: 80


## End of Analysis


In [44]:
# Let's look at August on a Sunday with a group size of 2
august_sunday_2 = df_split[
    (df_split["preferred_entry_date_month"] == "August")
    & (df_split["preferred_entry_date_day"] == "Sunday")
    & (df_split["minimum_acceptable_group_size"] == 2)
]

august_sunday_2_awarded_filter = august_sunday_2["awarded"] == True

august_sunday_2_awarded = august_sunday_2[august_sunday_2_awarded_filter]

# Let's look like July on a Sunday with a group size of 2
july_sunday_2 = df_split[
    (df_split["preferred_entry_date_month"] == "July")
    & (df_split["preferred_entry_date_day"] == "Sunday")
    & (df_split["minimum_acceptable_group_size"] == 2)
]

july_sunday_2_awarded_filter = july_sunday_2["awarded"] == True

july_sunday_2_awarded = july_sunday_2[july_sunday_2_awarded_filter]

# Let's look at September on a Sunday with a group size of 2
september_sunday_2 = df_split[
    (df_split["preferred_entry_date_month"] == "September")
    & (df_split["preferred_entry_date_day"] == "Sunday")
    & (df_split["minimum_acceptable_group_size"] == 2)
]

september_sunday_2_awarded_filter = september_sunday_2["awarded"] == True

september_sunday_2_awarded = september_sunday_2[september_sunday_2_awarded_filter]


# Lets look at september on a monday with a group size of 2
september_monday_2 = df_split[
    (df_split["preferred_entry_date_month"] == "September")
    & (df_split["preferred_entry_date_day"] == "Monday")
    & (df_split["minimum_acceptable_group_size"] == 2)
]

september_monday_2_awarded_filter = september_monday_2["awarded"] == True

september_monday_2_awarded = september_monday_2[september_monday_2_awarded_filter]

# Final all the entries in August, on a Tuesday, with a group size of 2
august_tuesday_2 = df_split[
    (df_split["preferred_entry_date_month"] == "August")
    & (df_split["preferred_entry_date_day"] == "Tuesday")
    & (df_split["minimum_acceptable_group_size"] == 2)
]

august_tuesday_2_awarded_filter = august_tuesday_2["awarded"] == True

august_tuesday_2_awarded = august_tuesday_2[august_tuesday_2_awarded_filter]


# Let's compare applying for permit in august on a tuesday for 2 with applying for permit in september on a monday for 2
print(
    f"Probability of being awarded a permit for August on a Tuesday with a group size of 2: {len(august_tuesday_2_awarded)/len(august_tuesday_2):.2%} ({len(august_tuesday_2_awarded)}/{len(august_tuesday_2)})"
)

print(
    f"Probability of being awarded a permit for August on a Sunday with a group size of 2: {len(august_sunday_2_awarded)/len(august_sunday_2):.2%} ({len(august_sunday_2_awarded)}/{len(august_sunday_2)})"
)

print(
    f"Probability of being awarded a permit for September on a Monday with a group size of 2: {len(september_monday_2_awarded)/len(september_monday_2):.2%} ({len(september_monday_2_awarded)}/{len(september_monday_2)})"
)

print(
    f"Probability of being awarded a permit for September on a Sunday with a group size of 2: {len(september_sunday_2_awarded)/len(september_sunday_2):.2%} ({len(september_sunday_2_awarded)}/{len(september_sunday_2)})"
)

print(
    f"Probability of being awarded a permit for July on a Sunday with a group size of 2: {len(july_sunday_2_awarded)/len(july_sunday_2):.2%} ({len(july_sunday_2_awarded)}/{len(july_sunday_2)})"
)

Probability of being awarded a permit for August on a Tuesday with a group size of 2: 1.67% (19/1136)
Probability of being awarded a permit for August on a Sunday with a group size of 2: 3.88% (22/567)
Probability of being awarded a permit for September on a Monday with a group size of 2: 1.53% (10/652)
Probability of being awarded a permit for September on a Sunday with a group size of 2: 4.60% (15/326)
Probability of being awarded a permit for July on a Sunday with a group size of 2: 4.94% (17/344)


The probabilities above should come as a shock to anyone who applied for a permit on Monday in September thinking they were making a safe bet! They could have traded it for a Tuesday in August and had similar odds amongst their peers. Or, really increased their odds by applying on a Sunday in August.

However, equally surprising is similar odds for a Sunday in September and a Sunday is July, holding group size constant.


## Key Takeaways

### General

-   July sees the second most application options entries by month (~40k for August, ~30k for July, ~23k for September).
-   May, October, and June had by far the best awarded-to-skipped ratios. There's not a huge difference between July and August/September. I don't think you gain that much by applying for September or July over August.
-   Sunday sees by far the least amount of competition.
-   Mondays are very competitive! Avoid Thursdays, Fridays, and Mondays.
-   Saturdays are less competitive than Mondays.
-   Colchuck isn't that much less competitive than Core.

### Core Zone

-   Applicants should submit _at least one_ option for the Core Enchantments out of their three options.
-   Friday, Tuesday, and Sunday had the best awarded-to-skipped ratios for the Core Enchantments
-   Group sizes of 1, 3, and 2 had the best awarded-to-skipped ratios for the Core Enchantments
-   Avoid high odd numbers for group sizes in the Core Enchantments. A group size of 8 is better than 5 and 7. These group sizes have the worse awarded-to-skipped ratios.

-   There's a bit of game theory going on with the lottery. Applicants should consider what other people do in terms of group size (odd vs. even) and day of the week.

## What I'd Do

-   If I had to apply today based on these results, this is what I'd do:
    -   Option 1: Core Enchantments Zone, Group Size of 3, Sunday, August
    -   Option 2: Colchuck Zone, Group Size of 3, Sunday, late July
    -   Option 3: Stuart Zone, Group Size of 3, Sunday, August

Let me explain each of these choices and my reasoning behind them.

### Core Enchantments Zone, Group Size of 3, Sunday, August

A total of 1,039 applicants were awarded permits other than the Core Zone **despite** applying for at least one Core Zone option (~42.5% of the permits awarded). If you add the people that applied for the Core Zone and won that number jumps to 1,712 (70%). Therefore, you don't lose much by applying for the Core Zone in one entry. You still have a good chance of winning a permit. What matters is that you hedge your chances of winning in later options.

Knowing that I should apply for the Core Zone, I'd reduce my chances of being skipped in two ways. First, I'd select a group size of 3. A group size of three had the second best awarded-to-skipped ratio which was ahead of a group size of 2. Group sizes of 4 had a worse ratio than group sizes of 6. A group size of 1 would be the best option, but I'd rather not go alone. Second, I'd choose a Sunday. Starting your trip on a Sunday had, by far, the best ratio of awarded-to-skipped. Trips starting on Friday or Monday were about twice as likely to be skipped rather than awarded compared to Sundays.

As for the month, the best months for avoiding being skipped are May, October, and June. Unfortunately, these months are all questionable weather months. They're not worth the effort or risk to me. Conversely, July, August, and September are better months for weather. So why would I choose August instead of July or September? Well, the month with the worse awarded-to-skipped ratio was September and the difference between the awarded-to-skipped ratio for July to August was pretty small. The benefits of August seem to greatly outweight the small gain of choosing July over August.

The choice of group size is only relevant for the Core Zone. The entry date of Sunday is repeated in all three options so I won't repeat my reasoning in later explanations.

### Colchuck Zone, Group Size of 4, Sunday, late July

The Snow Zone provided the second most permits in the lottery behind the Core Enchantment Zone. Additionally, applicants were more than twice as likely to receive a permit for the Snow Zone (7.46%) compared to the Colchuck Zone (2.97%). So why would I still choose the Colchuck Zone over the Snow Zone given these probabilities?

There are two reasons why applying for the Snow Zone is less desirable than the Colchuck Zone:

1. The hike into the Snow Zone is longer than the Colchuck Zone and the climb from the Snow Zone into the Lower Enchantments is worse, in my opinion, than the climb from Aasgard Pass (above Colchuck) to the Upper Enchantments. From the Snow Lakes trailhead into the Core Enchantments is some 8 or 9 miles if I remember. And the rock faces, big step up, and unneasy terrain between Snow Lakes and the Lower Enchantments is brutal and, _in my opinion_, more dangerous than Aasgard Pass. There's a reason much more people are going up Aasgard than Snow Lakes. It's enough for me to ignore the probability data.

2. Colchuck Lake is beautiful. Even if you don't decide to go up Aasgard Pass into the Upper Enchantments a trip to Colchuck Lake is stunning. Even with the crowds, crazy parking lots, etc. The scenery at Colchuck Lake is far and away more worth it than Snow Lake. The Snow Lakes are a loved part of the Enchantment Permit Zone, but they are not why people go to the Enchantment Permit Zone. I'll take my chances.

I'd be slightly increasing my chances of being awarded a permit compared to the Core Enchantment Zone in the Colchuck Zone in Option 2.

As for choosing late July, instead of August, is a break from my Option 1 explanation. What gives? Colchuck is a few thousand feet lower than the Core Enchantment Zone. The climb up would be easier and the conditions would _probably_ be better. I would take the small increase in awarded-to-skipped ratio in this situation because I'd be camping at a lower and easier to reach elevation. Camping at Colchuck Lake would still provide the opportunity to hike up to the Enchantments during the day.

Day selection are the same as Option 1

### Stuart Zone, Group Size 4, Sunday, August

The previous two options saw the zones and months change in an attempt to better my chances as my options get deeper. Option 3 is no exception. I again want to increase my chances by choosing a less competitive zone.

The zones of the Enchantments are hard to understand. Making things worse are "complete" guides online that don't talk about how different each zone is. For example, the Eightmile/Caroline Zone is no where near the beautiful lakelets and sharp peaks of the Core Enchantments or Colchuck Zone. The Jack Creek Fire burned the landscape around Eightmile Lake making it, as the numbers show, less desirable. The Forest Service has also had to reduce permits in the area due to the fire.

For the uninitiated, the Snow Zone and Colchuck may seem on level playing ground for having a chance to get up into the Core Enchantments but they are quite different in difficulty and beauty. There are reasons why the Colchuck Zone is almost as competitive as the Core Zone.

Likewise, the Stuart Zone is miles away from the Colchuck Zone and much different than all the other zones. Despite the Stuart Zone not having the most lakes, or jagged peaks, it's a very underrated zone. As mentioned in an online trip report, the lake is not dammed, giving it a very natural shoreline. It also, due to it's unpopularity, provides more solitude than the other zones while still providing pretty spectacular views of the surrounding mountains (including Mount Stuart).

The trail to Stuart Lake is certainly not as steep, rocky, or long as the trail to the other more popular zones.

Camping at Stuart Lake provides opportunities for a grueling obstacle course hike up to the magnificent Horseshoe Lake, or a worthy day hike to Colchuck Lake. It's not _THE_ Enchantments, but it's a great plan B.

The reasoning behind day of the week, and Month are the same as Option 1.


# Future Analysis Calculations

## Full Application

-   x Total applications
-   x Total awarded permits
-   x Probability of being awarded permit
-   x Total awarded permits by awarded status across different zones
-   x Probability you were awarded a permit given you applied for the Core Enchantments Zone
-   x Probability you were awarded a permit given you did not apply for the Core Enchantments Zone
-   x Probability that you were awarded a permit for the Core Enchantments
-   x Probability awarded permit for a zone given one of your application preferences were for that zone.
-   x Probability you were awarded Core Enchantments given all your entries were for Core Enchantment Zone.

## Split Entries

-   x Total entries by month
-   x Total entries by day
-   x Awarded permits by zone
-   x Skipped permits by zone
-   x AtS ratio by zone
-   x Awarded permits by month
-   x Skipped permits by month
-   x AtS ratio by month
-   x Awarded permits by day of week
-   x Skipped permits by day of week
-   x AtS ratio by day of week

### Core

-   x Awarded permits by month
-   x Skipped permits by month
-   x AtS ratio by month
-   x Awarded permits by day of week
-   x Skipped permits by day of week
-   x AtS ratio by day of week
-   x Total entries by group size
-   x Total awarded by group size
-   x Awarded permits by group size
-   x Skipped permits by group size
-   x AtS ratio by group size
-   x Total number of people awarded group size by day of week in August
